File: test2s/test2.py
import math
print("Hello!")
try:
    Input = float(input("Please enter a number: "))
except ValueError:
    print("Invalid input!")
    exit()

A = round(Input%2,10)
isOdd = False
if A == 0:
  print(f"The number is Even")
elif A == 1:
  print(f"The number is Odd")
  isOdd = True
else:
  print(f"The number is Decimal,Remainder: {A}")

isPositive = False
if Input > 0:
  rt = math.sqrt(Input)
  isPositive = True
  print(f"The number has two roots: {rt} and {-rt}")
elif Input == 0:
  print(f"The number has one root: 0")
else:
  print(f"The number doesnt have real root")
  rt = math.sqrt(-Input)
  print(f"Two of Imaginary roots: {rt}i and {-rt}i")
Int_Input = int(Input)
if Int_Input == 2:
  print("The number is a prime")
elif Int_Input == 1:
  print("The number is NOT a prime")
elif isOdd and isPositive:
  match = None
  for i in range(3,math.isqrt(Int_Input)+1,2):
    if Int_Input%i == 0:
      print("The number is NOT a prime")
      match = i
      break
  if match == None:
    print("The number is a prime")
else:
  print("The number is NOT a prime")

File: test2s/test2.lua
io.write("hello!\n")
io.write("Please Input a number: ")
local Input = tonumber(io.read())
local isOdd = false
if not Input then io.write("Please insert a number!") return end
io.write("\n")
local A = Input%2
io.write("The number is ")
if A == 0 then
  io.write("Even\n")
elseif A == 1 then
  io.write("Odd\n")
  isOdd = true
else
  io.write("Decimal,Remainder: " ..  A .. "\n")
end
local isPositive = false
io.write("The number has ")
if Input > 0 then
  local B = math.sqrt(Input)
  isPositive = true
  io.write("two roots: " .. B .. " and " .. -B  .."\n")
elseif Input == 0 then
  io.write("one root: 0\n")
else 
  local B = math.sqrt(-Input)
  io.write("no real root\n")
  io.write("Two of Imaginary roots: " .. B .. "i and " .. -B .. "i\n")
end
io.write("The number is ")
if Input == 2 then io.write("a prime\n")
elseif Input == 1 then io.write("NOT a prime\n")
elseif isOdd and isPositive then
  local match = nil
  for i = 3,math.ceil(math.sqrt(Input)),2 do
    if Input%i == 0 then
      io.write("NOT a prime\n")
      match = i
      break
    end
  end
  if match == nil then io.write("a prime\n") end
else io.write("NOT a prime\n") end
